key-uglifier/src/uglifier.rs

```diff
@@ -1,70 +1,55 @@
 use colored::Colorize;
 use hex::ToHex;
+use std::fmt::Write;
 
-fn param_set_default_encryption_key(key_length: u8) -> Vec<u8>
-{
-    let default_key = vec![0x62u8; key_length as usize];
-    return default_key
+fn param_set_default_encryption_key(key_length: u8) -> Vec<u8> {
+    vec![0x62u8; key_length as usize]
 }
 
-fn read_hex_nibble(c: char) -> (bool, String)
-{
-    if (c >='0') && (c <= '9'){
-        return (true, c.to_string());
-    }
-    else if (c >='A') && (c <= 'F')
-    {
-        return (true, c.to_string());
-    }
-    else if (c >='a') && (c <= 'f')
-    {
-        return (true, c.to_string());
-    }
-    else
-    {
+fn read_hex_nibble(c: char) -> (bool, String) {
+    if "0123456789ABCDEFabcdef".contains(c) {
+        (true, c.to_string())
+    } else {
         // printf("[%u] read_hex_nibble: Error char not in supported range",nodeId);
-        return (false, '0'.to_string().on_color("green").to_string());
+        (false, '0'.to_string().on_color("green").to_string())
     }
 }
 
-fn convert_to_hex(key_input: &String, key_length: u8) -> (bool, String) {
+fn convert_to_hex(key_input: &str, key_length: u8) -> (bool, String) {
     let mut key_output = String::new();
-    let mut bad_chars: bool = false;
+    let mut bad_chars = false;
 
-    for c in key_input.chars().enumerate(){
-        if c.0 > ((key_length as usize) * 2) - 1 {
+    for (index, character) in key_input.chars().enumerate() {
+        if index > ((key_length as usize) * 2) - 1 {
             break;
         }
-        let num = read_hex_nibble(c.1);
-        key_output.push_str(&num.1);
-        bad_chars |= !num.0;
+        let (ok, num) = read_hex_nibble(character);
+        key_output.push_str(&num);
+        bad_chars |= !ok;
```

Comment:
Personally, I would prefer consistent name of this boolean variable. It's hard to read this expression and keep in mind that _this function_ return boolean that indicates an error, but _that_ indicates ok.

By the way, returning `ok` as a separate bool variable may be replaced by the `Result`. For example:
```rust
fn read_hex_nibble(c: char) -> Result<char, String> {
//...
}

fn convert_to_hex() {
    // ...
    match read_hex_nibble(character) {
        Ok(c) => key_output.push(c),
        Err(err) => {
            key_output.push_str(err);
            bad_chars = true;
        }
    }
}
```